### PR TITLE
Processing data in models

### PR DESCRIPTION
**Pull request by Kakalanp**
# Changes in this PR:

I've added 6 methods to different files, mostly on models

## Methods in files:
- models:
  - user:
    - **most_recent_posts**: shows the 5 most recent posts
  - post:
    - **update_post_counter**: increments the number of posts for a given user
    - **most_recent_comments**: shows the 5 most recent posts
  - like:
    - **update_like_counter**: increments the number of likes for a given post
  - comment:
    - **update_comments_counter**: increments the number of comments for a given post
- controllers:
  - application_controller:
    - **current_user**: returns the first user from a database.

### CODE REVIEW

File: app/models/comment.rb [hunk 1]
```diff
@@ -1,4 +1,8 @@
 class Comment < ApplicationRecord
   belongs_to :user
   belongs_to :post
+
+  def update_comments_counter
+    post.increment!(:CommentsCounter)
+  end
```
Comment (rotimiazeez) [inline app/models/comment.rb]: - [x] I'd suggest you use *one of the [Rails Active Callbacks](https://guides.rubyonrails.org/active_record_callbacks.html) -* `after_save` to call the `#update_comments_counter` method every time a new post has been saved to database, then make the `update_comments_counter` method private. 
For example
```ruby
after_save :update_comments_counter

  private

def update_comments_counter
  post.increment!(:comments_counter)
end
```

File: app/models/post.rb [hunk 2]
```diff
@@ -2,4 +2,12 @@ class Post < ApplicationRecord
   belongs_to :user
   has_many :comments, dependent: :destroy
   has_many :likes, dependent: :destroy
+
+  def update_post_counter
+    user.increment!(:PostsCounter)
+  end
+
```
Comment (rotimiazeez) [inline app/models/post.rb]: - [x] I'd suggest you use *one of the [Rails Active Callbacks](https://guides.rubyonrails.org/active_record_callbacks.html) -* `after_save` to call the `#update_post_counter` method every time a new post has been saved to database, then make the `update_post_counter` method private. 
For example
```ruby
after_save :update_post_counter

  private

  def update_post_counter
    author.increment!(:posts_counter)
  end
```

File: app/models/like.rb [hunk 3]
```diff
@@ -1,4 +1,8 @@
 class Like < ApplicationRecord
   belongs_to :user
   belongs_to :post
+
+  def update_like_counter
+    post.increment!(:LikesCounter)
+  end
```
Comment (rotimiazeez) [inline app/models/like.rb:11]: - I'd suggest you use one of the [Rails Active Callbacks](https://guides.rubyonrails.org/active_record_callbacks.html) - `after_save` as stated above

**Review by rotimiazeez — CHANGES_REQUESTED** [review]
# Hi @Kakalanp 💪🏽 , 
## Good job so far! 👏🏽 
There are some issues that you still need to work on to go to the next project but you are almost there!
<FEEL FREE TO CHANGE THIS MESSAGE TO YOUR PERSONAL WORDS>

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/models/comment.rb (hunk 1 above)
Comment (Kakalanp) [reply app/models/comment.rb]: Thanks, I didn't know that! :+1:

**Review by SajjadAhmad14 — APPROVED** [review]
# Status: Approved :heavy_check_mark: :heavy_check_mark: :heavy_check_mark:

### Hi Joaquin, 
## Congratulations 🎉 Thank you for implementing all the requirements 🙏
### Please go ahead and merge the branch!

## Happy Coding 👏👏👏
#### Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
#### Please remember to tag me in your question so I can receive the notification.
As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

**Comment by Kakalanp** [comment]
Thank you! Merging right away :shipit:

**Pull request merged**